src/libsyntax/attr.rs

```diff
@@ -77,6 +81,93 @@ pub fn is_used(attr: &Attribute) -> bool {
     })
 }
 
+pub trait AttrMetaListItemMethods {
```

Comment:
Doesn't sounds like there is a good solution here, but at the least something needs renaming or there needs to be some docs here - totally unclear to me what the abstractions here represent